[MM-58004] Update logged fields of users

hanzei opened the pull request:
#### Summary

- Add `auth_data` to the logged user fiedls
- Remove some spammy and less useful fields like `bot_description`
- Fix access to pointer fields

Based on https://github.com/mattermost/mattermost/pull/26836

```json
{
    "timestamp": "2024-04-09 10:31:54.029 +02:00",
    "level": "debug",
    "msg": "Ldap sync foreign user",
    "caller": "ldap/ldap_sync_job.go:384",
    "worker_name": "EnterpriseLdapSync",
    "job_id": "snfcn3portg4zmnefycuf51ewo",
    "job_type": "ldap_sync",
    "job_create_at": "Apr  9 08:31:38.738",
    "ldap_user": {
        "allow_marketing": false,
        "auth_service": "ldap",
        "bot_description": "",
        "bot_last_icon_update": 0,
        "create_at": 0,
        "delete_at": 0,
        "disable_welcome_email": false,
        "email": "success+testone@simulator.amazonses.com",
        "email_verified": true,
        "failed_attempts": 0,
        "id": "",
        "is_bot": false,
        "last_activity_at": 0,
        "last_password_update": 0,
        "last_picture_update": 0,
        "locale": "",
        "mfa_active": false,
        "notify_props": null,
        "position": "Test1 Title",
        "props": null,
        "remote_id": null,
        "roles": "",
        "terms_of_service_create_at": 0,
        "terms_of_service_id": "",
        "timezone": null,
        "update_at": 0,
        "username": "test.one"
    }
}
```
#### Ticket Link
https://mattermost.atlassian.net/browse/MM-58004

#### Release Note
```release-note
Update logged fields of users
```

hanzei commented on server/public/model/user.go:
```diff
@@ -130,7 +130,7 @@ func (u *User) Auditable() map[string]interface{} {
 		"timezone":                   u.Timezone,
 		"mfa_active":                 u.MfaActive,
-		"remote_id":                  u.RemoteId,
+		"remote_id":                  SafeDereference(u.RemoteId),
```
Bonus bug fix :tada:

wiggin77 replied:
```suggestion
		"remote_id":                  u.GetRemoteID(),
```

wiggin77 commented on server/public/model/user.go:
```diff
@@ -142,7 +142,26 @@ func (u *User) Auditable() map[string]interface{} {
-	return u.Auditable()
+	return map[string]interface{}{
+		"id":              u.Id,
+		"create_at":       MillisToString(u.CreateAt),
```
This is a minor breaking change and makes the logging of this struct different than others in terms of timestamp format.  This will break any scripts or Prometheus queries.   I think this should be done on a major version and on all models at once, after consensus from R&D group.

hanzei replied:
Do we consider the logged messages part of an API that needs to be protected from breaking changes? Would removing fields also be considered a breaking change?

I'm leaning toward putting this change in the bugfix category as logs are designed to be consumed by humans, and UNIX millis are not human friendly.

> and makes the logging of this struct different than others in terms of timestamp format. This will break any scripts or 

I would love to also update the other structs :wink: There are only 6 of them.

hanzei replied:
One point to add: The `timestamp` field of the log entries also uses a formatted string, not unix millis.

wiggin77 replied:
I suggest checking with cloud team and/or SRE to see if this change would break anything in their world.

wiggin77 commented on server/public/model/user.go:
```diff
@@ -142,7 +142,26 @@ func (u *User) Auditable() map[string]interface{} {
+		"locale":          u.Locale,
+		"timezone":        u.Timezone,
+		"mfa_active":      u.MfaActive,
+		"remote_id":       SafeDereference(u.RemoteId),
```
```suggestion
		"remote_id":       u.GetRemoteID(),
```

wiggin77 reviewed (commented):
I have two simple change requests, and the rest looks good to me.  My only concern is whether this is considered a breaking change and I think we need more input for that.  I am also of two minds whether the timestamps in logs should be strings or millisecond ints.  I *often* just cut and paste those timestamps into queries, but I rarely try to mentally decode a human readable date from them.  I think more input should be collected for this as well.  @hanzei can you bring this up at the next dev meeting and let people weigh in?

hanzei commented:
@wiggin77 Heads up that I applied the same changes as in https://github.com/mattermost/mattermost/pull/26836: I removed the time stamp conversion and updated the branch.

PTAL

wiggin77 reviewed (approved):
LGTM :+1:
